flow/scene_update_context.cc

```diff
@@ -303,23 +310,14 @@ SceneUpdateContext::Frame::Frame(SceneUpdateContext& context,
       opacity_node_(context.session()),
       paint_bounds_(SkRect::MakeEmpty()),
       layer_(layer) {
-  const float depth = context.frame_physical_depth();
-  if (depth > -1 && world_elevation > depth) {
-    // TODO(mklim): Deal with bounds overflow more elegantly. We'd like to be
-    // able to have developers specify the behavior here to alternatives besides
-    // clamping, like normalization on some arbitrary curve.
-
-    // Clamp the local z coordinate at our max bound. Take into account the
-    // parent z position here to fix clamping in cases where the child is
-    // overflowing because of its parents.
-    const float parent_elevation = world_elevation - local_elevation;
-    local_elevation = depth - parent_elevation;
-  }
-  if (local_elevation != 0.0) {
-    entity_node().SetTranslation(0.f, 0.f, -local_elevation);
-  }
+  entity_node().SetLabel(label);
+  entity_node().SetTranslation(0.f, 0.f, z_translation);
   entity_node().AddChild(opacity_node_);
-  opacity_node_.SetOpacity(opacity_ / 255.0f);
+  // Scenic currently lacks an API to enable rendering of alpha channel; this only happens
+  // if there is a OpacityNode higher in the tree with opacity != 1. For now,
```

Comment:
Do you mean "this only happens ... with opacity == 1"? Clamping with 1 - epsilon seems to avoid `opacity == 1` case instead of `opacity != 1` case.

Reply:
Re-read it, the original language should be accurate. Tried to re-word it to make it clearer.